mytoyota/api.py

```diff
@@ -19,70 +23,107 @@ def uuid(self) -> str | None:
         """Returns uuid from controller"""
         return self.controller.uuid
 
-    async def set_vehicle_alias_endpoint(
-        self, new_alias: str, vehicle_id: int
-    ) -> dict[str, Any] | None:
-        """Set vehicle alias."""
-        return await self.controller.request(
+    async def set_vehicle_alias_endpoint(self, alias: str, guid: str, vin: str):

```

Comment:
You are using similar patterns for making HTTP requests. These patterns can be abstracted to reduce redundancy, e.g:

```python
    async def _request(self, method: str, endpoint: str, vin: str = None, **kwargs) -> Any:
        """Private helper to make HTTP requests."""
        headers = kwargs.pop('headers', {})
        if vin:
            headers['VIN'] = vin
        if method in ['PUT', 'POST']:
            headers['datetime'] = str(int(datetime.utcnow().timestamp() * 1000))
            headers['x-correlationid'] = str(uuid4())
            headers['Content-Type'] = 'application/json'
        return await self.controller.request(
            method=method,
            base_url=BASE_URL,
            endpoint=endpoint,
            headers=headers,
            **kwargs
        )
```

afterwads you cann call the function like this:

```python
    async def get_vehicle_status_endpoint(self, vin: str) -> dict[str, Any] | None:
        """Get information about the vehicle."""
        return await self._request("GET", "/v1/vehiclehealth/status", vin=vin)
```

Reply:
I had it this way originally that you could pass in the VIN until I worked out that some of the endpoints dont work if you pass "VIN" and require "vin" and some are the other way around! I can revisit this once the all the endpoints are settled on and if they all want lower case or upper case then can make the change.

Reply:
Vin is an optional argument in my example 😉 
But yes, let's keep it open until we have all the relevant endpoints.